src/emhass/csv_predictor.py

```diff
@@ -0,0 +1,135 @@
+        :type root: str
+        :param logger: The passed logger object
+        :type logger: logging.Logger
+        """
+        self.csv_file = csv_file
+        self.independent_variables = independent_variables
+        self.dependent_variable = dependent_variable
+        self.sklearn_model = sklearn_model
+        self.new_values = new_values
+        self.root = root
+        self.logger = logger
+        self.is_tuned = False
+
+    
+    def load_data(self) -> pd.DataFrame:
+        """Load the data."""
+        filename_path = pathlib.Path(self.root) / self.csv_file
+        if filename_path.is_file():
+            with open(filename_path, 'rb') as inp:
+                data = pd.read_csv(inp)
+        else:
+            self.logger.error("The cvs file was not found.")
+            raise ValueError(
+                f"The CSV file "+ self.csv_file +" was not found."
+            )
+
+        required_columns = self.independent_variables
+        
+        if not set(required_columns).issubset(data.columns):
+            raise ValueError(
+                f"CSV file should contain the following columns: {', '.join(required_columns)}"
+            )
+        return data
+    
+    def prepare_data(self, data) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Prepare the data.
+        
+        :param data: Input Data
+        :type data: pd.DataFrame
+        :return: A tuple containing the train data.
+        :rtype: Tuple[np.ndarray, np.ndarray]
+        
+        """
+        X = data[self.independent_variables].values
+        y = data[self.dependent_variable].values
+        X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
+        
+        return X_train, y_train
+    
+    
+    def predict(self) -> np.ndarray:
+        r"""The predict method to generate a forecast from a csv file.
+
+        :return: The np.ndarray containing the predicted value.
+        :rtype: np.ndarray
+        """
+        self.logger.info("Performing a prediction for "+self.csv_file)
+        # Preparing the data: adding exogenous features
+        data = self.load_data()
+        if data is not None:
+            X, y = self.prepare_data(data)
+        
+            if self.sklearn_model == 'LinearRegression':
+                base_model = LinearRegression()
+            elif self.sklearn_model == 'ElasticNet':
+                base_model = ElasticNet()
+            elif self.sklearn_model == 'KNeighborsRegressor':
+                base_model = KNeighborsRegressor()
+            else:
+                self.logger.error("Passed sklearn model "+self.sklearn_model+" is not valid")
+            # Define the forecaster object
+            self.forecaster = base_model
+            # Fit and time it
+            self.logger.info("Predict through a "+self.sklearn_model+" model")
+            start_time = time.time()
+            self.forecaster.fit(X, y)
```

Comment:
Creat a separate method for this, a `fit` method

Reply:
Do you mean I have to create a `def` for the fit method?
Does that `def` also needs a command_line `def` also?